flask_multipass_cern.py

```diff
@@ -170,20 +197,53 @@ def _fix_phone(self, data):
             return
         data['telephone1'] = self.settings['phone_prefix'] + phone
 
-    def _extract_extra_data(self, data):
-        return {'cern_person_id': data.pop('cernPersonId', None)}
+    def _extract_extra_data(self, data, default=None):
+        return {'cern_person_id': data.pop('cernPersonId', default)}
 
     def get_identity_from_auth(self, auth_info):
-        data = self._fetch_identity_data(auth_info)
+        upn = auth_info.data.get('cern_upn')
         groups = auth_info.data.get('groups')
+        cache_key_prefix = f'flask-multipass-cern:{self.name}'
+
         if groups is not None and self.cache:
             groups = {x.lower() for x in groups}
-            cache_key = f'flask-multipass-cern:{self.name}:groups:{data["upn"]}'
-            self.cache.set(cache_key, groups, 1800)
+            cache_key = f'{cache_key_prefix}:groups:{upn}'
+            self.cache.set(cache_key, groups, timeout=CACHE_LONG_TTL)
+            self.cache.set(f'{cache_key}:timestamp', datetime.now(), timeout=CACHE_TTL)
+
+        try:
+            data = self._fetch_identity_data(auth_info)
+
+            if self.cache:
+                phone = data.get('telephone1', None)
+                affiliation = data.get('instituteName', None)
```

Comment:
```suggestion
                phone = data.get('telephone1')
                affiliation = data.get('instituteName')
```

that's already the default